docs/docs/development/onboarding.md

```diff
@@ -0,0 +1,257 @@
+
+- [SCALE encoding (Spec)](https://spec.polkadot.network/id-cryptography-encoding#sect-scale-codec)
+- [SCALE Codec (Substrate)](https://docs.substrate.io/reference/scale-codec/)
+
+## 2.3. Runtime Internals
+
+### Extrinsics (Transactions) 💳
+
+- [Transaction Types](https://docs.substrate.io/learn/transaction-types/)
+- [Transaction Lifecycle](https://docs.substrate.io/learn/transaction-lifecycle/)
+
+### Weights & Fees ⚖️ 💸
+
+- [Transaction Weights (Gas) and Fees](https://docs.substrate.io/build/tx-weights-fees/)
+
+### Accounts, addresses, and keys 👤 🔑
+
+- [Accounts Addresses Keys](https://docs.substrate.io/learn/accounts-addresses-keys/)
+
+### Storage 💾
+
+Storing and retrieving data, key/value generation and types of storage values.
+
+- [State Transitions and Storage](https://docs.substrate.io/learn/state-transitions-and-storage/#querying-storage)
+- [Transactional Storage](https://docs.substrate.io/build/runtime-storage/#transactional-storage)
+
+### Pallets (Modules) 🧱
+
+Pallets communicate and interact with each other via events, storage, calls, hooks, etc.
+
+- [Pallet Coupling](https://docs.substrate.io/build/pallet-coupling/)
+- [Events and Errors](https://docs.substrate.io/build/events-and-errors/)
+
+# 3. Specifics of implementing a Runtime in Go 🔍🐛
+
+Developing a framework for writing Polkadot runtimes in Go is not a straight forward process, accompanied with many blockers and issues that need to be resolved. Most of the issues are related to the incompatibilities between the design decisions around the Polkadot protocol and the Go language. Here are some of the major challenges that we faced while working on the project:
+
+- Missing support for standalone Wasm (**MVP**) 🕳️
+- GC that is required to work with an external allocator, provided by the Host 💣
+- Immature toolchain based on [custom Tinygo](https://github.com/LimeChain/tinygo) 🐣
+- [SCALE codec](https://github.com/LimeChain/goscale) implementation with minimal reflection 🪩
+- Writing mostly low-level and unsafe Go (none of the concurrency capabilities are utilized) ⚠️
+- The spec lacks details regarding the Runtime; thus, you should be able to read Rust code, which is the actual source of truth. 🦀
+
+Most of the things are documented here [Gosemble Runtime Architecture](https://limechain.github.io/gosemble/overview/runtime-architecture/) , but more up to date version of the docs can be found [Gosemble docs](https://github.com/LimeChain/gosemble/tree/develop/docs/docs) 🚧.
+
+# 4. Tasks 📝
+
+### 4.1. Compile a Runtime from Gosemble and run it in a Substrate node
+
+1. Install `git` and `docker`
+2. Clone the Gosemble repo - `git clone https://github.com/LimeChain/gosemble.git`
+3. Checkout the development branch - `git checkout develop`
+4. Pull all necessary git submodules - `git submodule update --init --recursive`
+5. Build the runtime - `make build-docker-dev`
```

Comment:
`make build-docker-benchmarking` as there are tests related to the benchmarking tag in benchmarking folder